### PR TITLE
[CHK-3249] feat: add mock for IO app flow

### PR DESCRIPTION
**Pull request by scaminati-bv**
#### List of Changes
- add mock with node express
- include multiple outcome use cases based on transactionId suffix

#### Motivation and Context
This PR adds a mock with a list of use cases. 
Based on the rpt suffix (the list is in the readme), you can get redirect with a specific outcome.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,4 +59,35 @@ Test use cases:
...
+
+- *APP IO*
+    1. start script ```yarn start-io-mock``` on other terminal
+    2. open on browser http://localhost:1234/ecommerce-fe/esito#clientId=IO&sessionToken=test&transactionId=1234
+
+
+## IO Mock outcome result
+The ecommerce transaction get transaction endpoint `/checkout/webview/v1/transactions/:transactionId` is driven by the following mockFlow values:
+
+| MOCK FLOW                                          | RptId Suffix | OUTCOME              |
```
Comment (giovanniberti) [inline README.md:72]: Do you mean transaction id?
```suggestion
| MOCK FLOW                                          | Transaction Id Suffix | OUTCOME              | 
```

File: mock/io-mock.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  else if (req.params.transactionId.endsWith("78"))
+    res.send(mockTransactionData("EXPIRED", "NPG", "EXECUTED", "OK"));
+
+  else if (req.params.transactionId.endsWith("79"))
+    res.send(mockTransactionData("AUTHORIZATION_REQUESTED", "NPG", undefined, undefined));
+
+  else 
+    res.send(mockTransactionData("NOTIFIED_OK", "NPG", "EXECUTED", "OK"));
+
+});
```
Comment (giovanniberti) [inline mock/io-mock.js:72]: This if-else chain could probably be refactored as follows:

```js
const suffixMap = {
    "61": {
        status: "NOTIFICATION_REQUESTED",
        gateway: "NPG",
        gatewayAuthorizationStatus: "EXECUTED",
        outcome: "OK"
    },
    "62": {
        status: "NOTIFICATION_REQUESTED",
        gateway: "NPG",
        gatewayAuthorizationStatus: "EXECUTED",
        outcome: "KO"
    }
    // --snip--
};

const suffix = transactionId.slice(-2);
const params = suffixMap[suffix];

res.send(mockTransactionData(params.status, params.gateway, params.gatewayAuthorizationStatus, params.outcome));
```

What do you think?

**Review by giovanniberti — APPROVED** [review]

File: mock/io-mock.js (hunk 2 above)
Comment (scaminati-bv) [reply mock/io-mock.js:72]: yes, it is more readable

Comment (scaminati-bv) [reply mock/io-mock.js]: done e05a547cab8f75ea174150eac4505b1cce3eb87d

Comment (scaminati-bv) [reply mock/io-mock.js]: 787189dc16b1e507cbbdfbeaf7f65490cce428e2 here i fix a log syntax error on previous commit

**Pull request merged**